mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/plan/synchronization/IndexingPlanSynchronizationStrategyConfigurationContext.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.mapper.pojo.plan.synchronization;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Consumer;
+
+import org.hibernate.search.engine.backend.work.execution.DocumentCommitStrategy;
+import org.hibernate.search.engine.backend.work.execution.DocumentRefreshStrategy;
+import org.hibernate.search.engine.backend.work.execution.OperationSubmitter;
+import org.hibernate.search.engine.reporting.FailureHandler;
+import org.hibernate.search.mapper.pojo.work.SearchIndexingPlanExecutionReport;
+import org.hibernate.search.util.common.annotation.Incubating;
+
+@Incubating
+public interface IndexingPlanSynchronizationStrategyConfigurationContext {
+
+	/**
+	 * @param strategy A strategy describing how commits should be handled after document changes are applied.
+	 * Defaults to {@link DocumentCommitStrategy#NONE}.
+	 */
+	void documentCommitStrategy(DocumentCommitStrategy strategy);
+
+	/**
+	 * @param strategy A strategy describing how refresh should be handled after document changes are applied.
+	 * Defaults to {@link DocumentRefreshStrategy#NONE}.
+	 */
+	void documentRefreshStrategy(DocumentRefreshStrategy strategy);
+
+	/**
+	 * Set the handler for the (asynchronous) indexing future.
+	 * <p>
+	 * This typically involves waiting on the given future,
+	 * to prevent the thread from resuming execution until indexing is complete.
+	 *
+	 * @param handler A handler that will be passed a future representing the progress of indexing.
+	 * Defaults to a no-op handler.
+	 * The future will be completed with an execution report once all document changes are applied.
+	 * If any document change or the commit/refresh required by{@link #documentCommitStrategy(DocumentCommitStrategy)}
+	 * and {@link #documentRefreshStrategy(DocumentRefreshStrategy)} failed,
+	 * the report will {@link SearchIndexingPlanExecutionReport#throwable() contain a throwable}
+	 * and (if applicable) {@link SearchIndexingPlanExecutionReport#failingEntities() a list of failing entities}.
+	 */
+	void indexingFutureHandler(Consumer<CompletableFuture<SearchIndexingPlanExecutionReport>> handler);
```

Comment:
Could you use this signature, so that we easily can add generics to `SearchIndexingPlanExecutionReport` in the future?

Yes it's complex to the point of being obscure, but this is a rather advanced API anyway, so that's no big deal.

```suggestion
	void indexingFutureHandler(Consumer<? super CompletableFuture<? extends SearchIndexingPlanExecutionReport>> handler);
```